src/v/kafka/server/quota_manager.cc

```diff
@@ -36,55 +55,114 @@ quota_manager::quota_manager()
       config::shard_local_cfg().kafka_client_group_byte_rate_quota.bind())
   , _target_fetch_tp_rate_per_client_group(
       config::shard_local_cfg().kafka_client_group_fetch_byte_rate_quota.bind())
+  , _client_quotas{client_quotas}
   , _gc_freq(config::shard_local_cfg().quota_manager_gc_sec())
   , _max_delay(config::shard_local_cfg().max_kafka_throttle_delay_ms.bind()) {
-    _gc_timer.set_callback([this] {
-        auto full_window = _default_num_windows() * _default_window_width();
-        gc(full_window);
-    });
+    _gc_timer.set_callback([this]() { gc(); });
 }
 
 quota_manager::~quota_manager() { _gc_timer.cancel(); }
 
 ss::future<> quota_manager::stop() {
     _gc_timer.cancel();
-    return ss::make_ready_future<>();
+    co_await _gate.close();
+    co_return co_await ss::make_ready_future<>();
 }
 
 ss::future<> quota_manager::start() {
+    co_await seastar::smp::submit_to(
+      _client_quotas.shard_id(),
+      [_client_quotas = std::ref(_client_quotas)]() -> ss::future<> {
+          co_await _client_quotas.get().reset(client_quotas_map_t{});
+          co_return;
+      });
+
     _gc_timer.arm_periodic(_gc_freq);
-    return ss::make_ready_future<>();
+    co_return co_await ss::make_ready_future<>();
```

Comment:
`start` is called on every shard, so I think it's best for the other shards to guard the whole contents of this function by the worker shard.